woocommerce/woocommerce-ios
# Coupons: Sync details from remote and update apply rules

Part of #5765 


### Description

This PR updates coupon details screen with 2 things:
- Trigger remote syncing to make sure that coupon details are up-to-date.
- Localize content for the "Apply to" field. This takes into consideration different cases of apply rules: 
  - When only specific products or categories are defined: Display "x Products" or "x Categories"
  - When specific products/categories and exceptions are defined: Display "x Products except y Categories" etc.
  - When both specific products and categories are defined: Display "x Products and y Categories"
  - When only exceptions are defined: Display "All except x Products" or "All except y Categories"

### Testing instructions

1. Test the "Apply To" field:
- On WP Admin, navigate to Marketing > Coupons > Create coupons with different usage restrictions
- On the app, navigate to Menu > Coupons > select the created coupons. Notice that the "Apply To" field is as expected.

2. Test synchronizing:
- On the app, navigate to Menu > Coupons to view the list of coupons for your store.
- On WP Admin, navigate to Marketing > Coupons > select a coupon and edit some of its details.
- On the app, select the edited coupon from the list. Notice that the display details reflect the updated information correctly.

### Screenshots

| Single rule | Combined rule | Exception rule |
| ----- | ----- | ----- |
| <img src="https://user-images.githubusercontent.com/5533851/152932970-e795b8d7-ff68-4d33-84bc-02d2e1c61acb.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/152932549-8e3079f8-879b-4ecb-9502-8ca8b848dd91.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/152932714-05f40cc3-16b8-49ac-9497-7763370ccf18.png" width=320 /> |

---
